### PR TITLE
[HOPSWORKS-1374] Add Hudi API to HSFS and commits to the FeatureStore metadata

### PR DESCRIPTION
**Pull request by davitbzh**

### CODE REVIEW

**Comment by moritzmeister** [comment]
Don't forget to add the UNDO operations for rolling back.

File: files/default/sql/ddl/2.0.0__initial_tables.sql [hunk 1]
```diff
@@ -1901,6 +1902,7 @@ CREATE TABLE IF NOT EXISTS `cached_feature_group` (
   `offline_feature_group`          BIGINT(20)      NOT NULL,
   `online_enabled`                 TINYINT(1)      NULL,
   `default_storage`                TINYINT         NULL,
+  `timetravel_format`              ENUM('HUDI', 'NONE') NOT NULL DEFAULT 'HUDI',
```
Comment (SirOibaf) [inline files/default/sql/ddl/2.0.0__initial_tables.sql:1905]: This should be an `INT`. The persistence in hopsworks-ee takes care of doing the conversion.

File: files/default/sql/ddl/2.0.0__initial_tables.sql [hunk 2]
```diff
@@ -1584,7 +1585,7 @@ CREATE TABLE `training_dataset_join_condition` (
 /*!40101 SET character_set_client = @saved_cs_client */;
 
 --
--- Table structure for table `feature_store_feature`
+-- Table structure for table `on_demand_feature`
```
Comment (SirOibaf) [inline files/default/sql/ddl/2.0.0__initial_tables.sql:1588]: leftover change.

File: files/default/sql/ddl/2.0.0__initial_tables.sql [hunk 3]
```diff
@@ -1991,3 +1993,27 @@ CREATE TABLE `remote_group_project_mapping` (
...
+  `partition_id`                      BIGINT(20)      NOT NULL,
+  PRIMARY KEY (`feature_group_id`, `commit_id`),
+  KEY `commit_id_idx` (`commit_id`),
+  KEY `commit_date_idx` (`committed_on`),
+  CONSTRAINT `feature_group_fk` FOREIGN KEY (`feature_group_id`) REFERENCES `feature_group` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION,
+  CONSTRAINT `hopsfs_parquet_inode_fk` FOREIGN KEY (`inode_pid`, `inode_name`, `partition_id`) REFERENCES `hops`.`hdfs_inodes` (`parent_id`, `name`, `partition_id`)
+    ON DELETE CASCADE
+    ON UPDATE NO ACTION
+
+) ENGINE=ndbcluster DEFAULT CHARSET=latin1 COLLATE=latin1_general_cs;
```
Comment (SirOibaf) [inline files/default/sql/ddl/2.0.0__initial_tables.sql:2018]: Please fix the formatting here.

**Review by SirOibaf — CHANGES_REQUESTED** [review]

**Review by SirOibaf — APPROVED** [review]

**Pull request merged**